src/test/java/com/fasterxml/jackson/core/util/TestTextBuffer.java

```diff
@@ -192,4 +196,62 @@ public void testGetSizeFinishCurrentSegmentAndResetWith() throws Exception {
         assertEquals(2, textBuffer.size());
     }
 
+    public void testEmptyContentsAsFloat() {
+        TextBuffer textBuffer = new TextBuffer(null);
+        assertThrows(NumberFormatException.class, () -> {
+            textBuffer.contentsAsFloat(false);
+        });
+    }
+
+    public void testEmptyContentsAsFloatFastParser() {
+        TextBuffer textBuffer = new TextBuffer(null);
+        assertThrows(NumberFormatException.class, () -> {
+            textBuffer.contentsAsFloat(true);
+        });
+    }
+
+    public void testContentsAsFloat() throws IOException {
+        TextBuffer textBuffer = new TextBuffer(null);
+        textBuffer.resetWithString("1.2345678");
+        assertEquals(1.2345678f,  textBuffer.contentsAsFloat(false));
+    }
+
+    public void testContentsAsFloatFastParser() throws IOException {
+        TextBuffer textBuffer = new TextBuffer(null);
+        textBuffer.resetWithString("1.2345678");
+        assertEquals(1.2345678f,  textBuffer.contentsAsFloat(true));
+    }
+
+    public void testEmptyContentsAsDouble() {
+        TextBuffer textBuffer = new TextBuffer(null);
```

Comment:
Not legal test as per usage patterns: not just for empty case but more generally, parsers will tokenize numbers using JSON (etc) pattern that describes valid numbers.
This simplifies decoding in cases decoder knows it need not validate input (such as our optimized `int`/`long` decoding logic).

Reply:
If `new TextBuffer(null)` is illegal, can we update the constructor to throw an exception?

Reply:
Yes, that'd make sense

Reply:
Actually no n/m, that's `BufferRecycler` -- `null` is accepted.